server/src/main/java/de/uftos/builders/SpecificationBuilder.java

```diff
@@ -64,8 +67,41 @@ public SpecificationBuilder<T> optionalAndJoinIn(Optional<String[]> attributeVal
     }
     specification = specification.and(
         (root, query, cb) -> root.join(relationName).get(attributeName)
-            .in(List.of(attributeValue.get()))
+            .in((Object[]) attributeValue.get())
```

Comment:
Why does this need to be casted to an `Object[]`?

Reply:
Before I was passing a List of `String[]` which is wrong. Now I'm passing an array of `Object`. The cast was needed because the call to the method `in` was ambigous and it was not clear whether the varargs a non-varargs call was desired. I have since replaced it with something better.